### DISCUSSION TITLE
Welcome to CoolProp Discussions!

### DISCUSSION THREAD

**Discussion by thorade** [CoolProp/CoolProp#2206] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by thorade** [comment]
Enabling discussions feature on this github repository, as requested on the Google Group:
https://groups.google.com/g/coolprop-users/c/l_aY4vTT58Q

**Reply by ibell** [reply]
Do you want to introduce yourself too @thorade ?

**Reply by thorade** [reply]
Did that now, the suggestion was in the github default template, but actually it is quite nice to get to know everyone a bit!

**Comment by CSTritt2019** [comment]
I'm Charles. I'm a biomedical engineering professor at MSOE University in Milwaukee, Wisconsin, USA. I'm responsible for and often teach our bio-thermal-fluids courses (this was a two-quarter sequence, but our university switching to semesters after this year after which it will be a single course).  We use CoolProp for fluid properties (mostly just water/steam/moist air). We've traditionally used Matlab as our programming language but will be switching to Python with our switch to semesters.

**Comment by ibell** [comment]
I'm Ian Bell, self-appointed benevolent dictator for life ([BDFL](https://en.wikipedia.org/wiki/Benevolent_dictator_for_life)) of CoolProp.  I started the work on CoolProp when I was a graduate student at Purdue University. I continued the development while a post-doc at the University of Liege during which time CoolProp really reached full maturity. The quantum leap from version 4 to version 5 in the change of the core architecture could not have happened without the hard work of @jowr. Since my time in Belgium I work at the National Institute of Standards and Technology, in the team that develops REFPROP. I don't work on CoolProp anymore on a day-to-day basis, but others from the community are starting to really step up, and more of that is needed for CoolProp to continue to thrive.

**Comment by henningjp** [comment]
I'm Jeff, a Chemical Engineer and thermodynamics hobbyist (outside of my "real" job).  I got involved with CoolProp in 2016 and revived the ailing wrapper for legacy Mathcad (restored in v6.0) and now Mathcad Prime as well as some maintenance on other wrappers, CoolProp's GitHub documentation (and Wiki for noob GitHub users like me), and issue/PR templates.  Along the way, I fleshed out more of the IAPWS-IF97 backend (CoolProp/IF97) to CoolProp (which @ibell started) for more rapid water properties calculations and the ability to write IF97-only wrappers for power industry users who only really need to do fast calculations with water with a reduced memory footprint (vice the CoolProp interface).  It's been a great journey and I've learned a LOT from this community, especially @ibell .  I'm grateful for the opportunity to contribute.

**Comment by thorade** [comment]
I'm Matthis, I studied mechanical engineering in Hamburg, Germany, then did a PhD on heat exchangers for geothermal power plants at the German national lab for geosciences in Potsdam/Berlin, 2009-2014. As part of that work, I implemented a fluid properties library in Modelica, roughly at the same time as Ian implemented CoolProp. If CoolProp had existed back then, I would probably have used that instead. I never contributed any C++ code to CoolProp, but I did and do enjoy participating in some discussions, especially related to dynamic simulation and partial derivatives. After the PhD I worked for some years in building energy performance simulations (@ibpsa) at the university (@UdK-VPT), and since 2018 I work at Modelon (@modelon) as a simulation engineer, developing Modelica (@modelica) libraries for simulation and research of thermodynamic cycles like heat pumps, refrigerators, air-conditioning, energy storage, etc. With a full time job and three kids, there is not much free time left to contribute to open-source libraries anymore :-(

**Comment by RJCFYV** [comment]
I'm an adjunct thermal sciences faculty member at U of Arkansas.  How hard would it be to create a stand-alone workbook, vs an add-in, with the CoolProp thermo property functions in VBA modules?  In a thermo or thermal modeling class, students often do work at home and then continue in a computer lab on campus.  To do so, the add-in must be installed on the machines in the computer lab, something our IT person is reluctant to do.  A few years ago in a NSF project at another university, Excel thermo property functions for water and R134a were made available as an add-in.  In that case, I was able to do the conversion to a stand-alone workbook.  Is that possible for CoolProp?

**Reply by CSTritt2019** [reply]
I'm just user of CoolProp, rather than a developer, so I don't have any particular insight into this. Good luck.

**Reply by henningjp** [reply]
@RJCFYV Sorry, this is question is old, but we must've missed that there was a question here.  CoolProp is quite extensive with many tens of thousands of lines of C++ code, a lot of it in external C++ repositories, that are compiled to a DLL.  The VBA functions make calls to that DLL.  Converting the whole thing to a VBA module that could be contained in a workbook (I've written some of those) is just not something that could be done (or that anyone would want to do).

**Reply by stephen-young** [reply]
Giving the students the DLL with the workbook could work, if you are able to do so. In my experience, calls to CoolProp functions in Excel usually work when the DLL is in the same folder as the spreadsheet (not guaranteed though). 

Another option is to use CoolProp to generate property tables and have VBA macros that perform some kind of interpolation on those tables so that the workbook will still work without access to CoolProp.

**Comment by Gualcar** [comment]
Greetings!

I'm Gustavo, a PhD student of Petroleum Engineering at University of Tulsa. I'm from Brazil and there I did my undergrad and masters in Chemical Engineer. Last year I worked as a scientific developer developing models for CO2 two-phase flow to be implemented in a 1D transient flow software, then I used CoolProp (python wrapper) to study pure CO2 properties and cross-validate my implementations of thermodynamic derivatives.

I'm really willing to contribute to the CoolProp community through discussions and code implementations (I know both Python and C++).  I already implemented cubic EOSs and flash routines for mixtures, and soon I will start to implement a library to calculate black-oil properties.

Do you need any help with coding? I tried to start the project but I got lost with so many flags in the CMakeLists file 😄

**Comment by mojtaba136411** [comment]


**Reply by ibell** [reply]
Please open a new discussion, this thread is for introduction

**Reply by mojtaba136411** [reply]
OK;
https://github.com/CoolProp/CoolProp/discussions/2316

**Comment by mojtaba136411** [comment]


**Reply by henningjp** [reply]
Please open a new discussion, this thread is for introduction

**Comment by danielkwalsh** [comment]
Hey, my name is Daniel Walsh. I studied physics at UC Santa Barbara, and then continued my studies as a PhD student in theoretical plasma physics at UC San Diego. My PhD work involves the frequencies of waves in strongly magnetized, non-neutral plasmas, and in particular their dependence on temperature and finite length. After graduating, I put my thermal and computational skills to work, and started a company with a few of my UCSB friends where we use thermodynamics to create physically-informed control systems to efficiently and scalably operate industrial cold storage facilities (industrial-scale food and beverage processing and storage). It's a bit of a mouthful, but that's a small price to pay for doing something that has a tangible positive impact on our world by reducing greenhouse gasses, easing load on the electrical grid, and improving the robustness of the food chain -- all while saving our customers money.

In my time away from work, I enjoy walking around cozy downtown Santa Barbara streets, creating YouTube video content to visually explain mathematics and physics, and raising monarch butterflies.

**Comment by dvd101x** [comment]
Hi, I'm David Contreras, a mechanical design engineer from Monterrey, Mexico. I have a background in Mechanical Engineering and a Master's in Manufacturing Systems, both from ITESM. During my studies, I had access to powerful engineering and thermodynamics software, which I missed after graduating.

That's why I discovered CoolProp for its thermodynamics capabilities. I started learning Python specifically to leverage CoolProp, and now I use it for experimental heat exchanger calculations at Güntner.

Beyond work, I'm passionate about CoolProp's pyodide implementation and building projects with its JavaScript wrapper. I'm also exploring ways to integrate it with online spreadsheets, although my programming skills are still developing, so my contributions are occasional.

My primary motivation is to make numerical computing more accessible to engineers. I believe this will bridge the gap between theoretical science and practical engineering, ultimately leading to more efficient real-world machines.

**Comment by cooper-david-a** [comment]
Hello, I'm Dave Cooper, a mechanical engineer with UAB.  I was an adjunct teaching thermal fluid systems for a while.  Now, still with UAB I work on a team developing -80 freezers for the I.S.S.

I made a silly little thermo property calculator (https://daveisthebest.com/ThermoPropertyCalculator/) using coolprop and was toying with the idea of expanding it to have similar capabilities as fchartsoftware's EES program.

**Reply by dvd101x** [reply]
Hi Dave, that's a neat example!

I'm also trying to have similar capabilities to EES.
- https://github.com/dvd101x/web-thermodynamics

**Reply by cooper-david-a** [reply]
Yeah, I was checking that out earlier.  Pretty cool.

**Comment by stephen-young** [comment]
Hey, I'm Stephen Young and I am a PhD student that is part of a group at Laurentian University in Sudbury, Ontario, Canada researching methods to reduce the energy demands of the Mining industry with a particular focus on compressed air production. Undergraduate training is in Mining Engineering but my graduate training has been more in the Chemical/Mechanical/Process engineering space (undergraduate, Masters and PhD all at Laurentian). CoolProp is a great tool that helped me a lot with my Masters research developing a 1-D hydrodynamic model for bubbly flows. I've mainly programmed in MATLAB but I do have some basic knowledge of Python, VBA and others.

**Comment by KennethGuier** [comment]
Hello, I'm Kenneth Guier and I work for an Industrial Refrigeration company in the UK. We have been using CoolProps since 2014. This year we are looking at absorption systems with water & ammonia mixtures. Now, from other discussions, we must buy the NIST 10, which also has properties for refrigerants and the like that CoolProps has. Now my manager is asking: Why should we stick to CoolProps if we get the same with NIST RefProps + Water & Ammonia mixtures for absorption systems?

Can you please help me answer?

**Reply by ibell** [reply]
As a developer of both libraries, I think the answer is that you don't have to choose A or B since CoolProp is free and supercharges REFPROP. I personally really like the interface of CoolProp (that I developed it has nothing to do with it :) ) and the use of CoolProp to call REFPROP also avoids a few families of bugs in REFPROP, particularly related to the use of mass fractions in the new high-level interface.

**Reply by henningjp** [reply]
I've answered the same question at work.  In Addition to the interface benefits, here are a few more:
1. CoolProp has an IF97 backend for water (faster for computations) and REFPROP uses the IAPWS-95 formulation for scientific use (more accurate, but slower).
2. CoolProp has an incompressible backend for aqueous solutions.
3. CoolProp has one of the only Humid Air formulations available.

**Reply by KennethGuier** [reply]
Great answers!

I didn't have any experience with REFPROP. Our colleagues at R&D for compressors use it with MS EXCEL.

We, in product development and system design, prefer Prime, Matlab, and Python.

Now, if anyone else wants to add to the above, please keep it coming!

Thank you!
